### PR TITLE
Camel-15211: add props to configure SSL context parameters

### PR DESCRIPTION
**Pull request by jono-morris**
Added  `camel.main.ssl.` parameters allowing Camel's SSL security parameters to be configured.

Tests added to camel-jetty, camel-netty, and camel-vertx.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: core/camel-main/src/main/java/org/apache/camel/main/DefaultConfigurationProperties.java [hunk 1]
```diff
@@ -159,6 +159,15 @@ public abstract class DefaultConfigurationProperties<T> {
     private String startupRecorderProfile = "default";
     private long startupRecorderDuration;
     private String startupRecorderDir;
+    private boolean sslEnabled;
+    private String sslKeyStore;
+    @Metadata(defaultValue = "changeit")
+    private String sslKeystorePassword = "changeit";
```
Comment (davsclaus) [inline core/camel-main/src/main/java/org/apache/camel/main/DefaultConfigurationProperties.java:165]: We should not have default passwords

**Review by davsclaus — CHANGES_REQUESTED** [review]
Add a new camel.main.xxx - should also update in PrepareCamelMainMojo and there is also code changes needed in BaseMainSupport

**Comment by jono-morris** [comment]
Thanks for the review.  Give me a day to make the changes.

**Comment by jono-morris** [comment]
Added camel.main.ssl.xxx, and updated PrepareCamelMainMojo and BaseMainSupport. 
Thanks.

**Comment by davsclaus** [comment]
Can we change `camel.main.ssl` to `camel.ssl` as this is not tied to main only.

**Comment by jono-morris** [comment]
Yes of course. Let me do that.

**Comment by jono-morris** [comment]
Changes committed changing `camel.main.ssl` to `camel.ssl`.  Thanks

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
We also need some work for camel-spring-boot, to make camel.ssl configuration possible via its auto configuration.

Its something like threadpool but not exactly the same, but you can take a look
https://github.com/apache/camel-spring-boot/tree/main/core/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/threadpool

**Comment by jono-morris** [comment]
Yes certainly, let me take a look.

**Pull request merged**